recipes/dbus/1.x.x/conanfile.py

```diff
@@ -54,69 +66,118 @@ def configure(self):
+            tc.project_options["selinux"] = "enabled" if self.options.get_safe("with_selinux", False) else "disabled"
+            tc.project_options["systemd"] = "enabled" if self.options.get_safe("with_systemd", False) else "disabled"
+            if self.options.get_safe("with_systemd", False):
+                tc.project_options["systemd_system_unitdir"] = os.path.join(self.package_folder, "lib", "systemd", "system")
+                tc.project_options["systemd_user_unitdir"] = os.path.join(self.package_folder, "lib", "systemd", "user")
+            if is_apple_os(self):
+                tc.project_options["launchd_agent_dir"] = os.path.join(self.package_folder, "res", "LaunchAgents")
+            tc.project_options["x11_autolaunch"] = "enabled" if self.options.get_safe("with_x11", False) else "disabled"
+            tc.project_options["xml_docs"] = "disabled"
+            tc.generate()
+            env = VirtualBuildEnv(self)
+            env.generate(scope="build")
+        else:
+            tc = CMakeToolchain(self)
+            tc.variables["DBUS_BUILD_TESTS"] = False
+            tc.variables["DBUS_ENABLE_DOXYGEN_DOCS"] = False
+            tc.variables["DBUS_ENABLE_XML_DOCS"] = False
+            tc.variables["DBUS_BUILD_X11"] = bool(self.options.get_safe("with_x11", False))
+            tc.variables["ENABLE_SYSTEMD"] = "ON" if self.options.get_safe("with_systemd", False) else "OFF"
+            tc.variables["DBUS_WITH_GLIB"] = bool(self.options.get_safe("with_glib", False))
+            tc.variables["DBUS_DISABLE_ASSERT"] = is_apple_os(self)
+            tc.variables["DBUS_DISABLE_CHECKS"] = False
+
+            # Conan does not provide an EXPAT_LIBRARIES CMake variable for the Expat library.
+            # Define EXPAT_LIBRARIES to be the expat::expat target provided by Conan to fix linking.
+            tc.variables["EXPAT_LIBRARIES"] = "expat::expat"
+
+            # https://github.com/freedesktop/dbus/commit/e827309976cab94c806fda20013915f1db2d4f5a
+            tc.variables["DBUS_SESSION_SOCKET_DIR"] = str(self.options.session_socket_dir)
+            tc.generate()
+            cmake_deps = CMakeDeps(self)
+            cmake_deps.generate()
+        pkg_config_deps = PkgConfigDeps(self)
+        pkg_config_deps.generate()
 
     def _patch_sources(self):
         apply_conandata_patches(self)
-        # Unfortunately, there is currently no other way to force disable
-        # CMAKE_FIND_PACKAGE_PREFER_CONFIG ON in CMake conan_toolchain.
-        replace_in_file(
-            self,
-            os.path.join(self.generators_folder, "conan_toolchain.cmake"),
-            "set(CMAKE_FIND_PACKAGE_PREFER_CONFIG ON)",
-            "",
-            strict=False,
-        )
+        if not self._meson_available:
+            # Unfortunately, there is currently no other way to force disable
+            # CMAKE_FIND_PACKAGE_PREFER_CONFIG ON in CMake conan_toolchain.
```

Comment:
cache variable?

Reply:
That appears to work, so I've updated this to use the cache variable.